src/units/attack_type.cpp

```diff
@@ -97,6 +97,157 @@ std::string attack_type::accuracy_parry_description() const
 	return s.str();
 }
 
+static bool bool_matches_if_present(const config& filter, const config& cfg, const std::string& attribute, bool def)
+{
+	if(filter[attribute].empty()) {
+		return true;
+	}
+
+	return filter[attribute].to_bool() == cfg[attribute].to_bool(def);
```

Comment:
When it was only in one file, having these as static functions seemed OK, but I'd prefer to have a utility library instead of multiple copies of these functions.

Reply:
i pushed https://github.com/wesnoth/wesnoth/pull/7814/commits/b8f7d61c7e8e92309924a97c392d44c425a3d4fa

static function remved

Reply:
@stevecotton i movedthese static in namespace in abilities.cpp in https://github.com/wesnoth/wesnoth/pull/7814/commits/1126451791374a2766c5067ab6fc22576a9fc1de commit